src/audio/pipeline/pipeline-params.c

```diff
@@ -290,6 +290,10 @@ static int pipeline_comp_prepare(struct comp_dev *current,
 		 dev_comp_id(current), dir);
 
 	if (!comp_is_single_pipeline(current, ppl_data->start)) {
+		/* ipc4 module is only prepared in its parent pipeline */
```

Comment:
@RanderWang can you please elaborate on what this means? Does this check if the MOD_ID has already been assigned for the current comp under consideration?

Reply:
@ranj063   pipeline is the context maintainer of its modules, .e.g. memory allocation.  In ipc4, each pipeline will be prepared and its modules are prepared at this time. we never prepare modules in other pipeline, only in its parent pipeline.

Reply:
Thanks @RanderWang what is a parent pipeline? In a pipeline with 2 copiers host and dai copier belonging to 2 different pipelines, which one is the parent and why?

Reply:
BTW im not questioning the logic here. It does fix the issue with capture but I would like to understand it better

Reply:
@ranj063  NP. when a copier is created, one param is pipeline id, please check following ipc msg. This pipeline is the parent.

````c
		struct {
			/**< Size of Data::param_data[] (in dwords) */
			uint32_t param_block_size   : 16;
			/**< ID of module instance's parent pipeline */
			uint32_t ppl_instance_id    : 8;
			/**< ID of core that instance will run on */
			uint32_t core_id            : 4;
			/**< Processing domain, 0-LL, 1-DP */
			uint32_t proc_domain        : 1;
			/* reserved in cAVS  */
			uint32_t extended_init      : 1;
			uint32_t _hw_reserved_2     : 2;
		} r;
````

Reply:
More: for IPC3 case with multiple pipelines, only first pipeline is prepared and every component is prepared along the path from host to dai even they are in different pipelines. But it is not true for IPC4, each pipeline is prepared by ipc message.

Reply:
@RanderWang thanks, so what does this check do if (IPC4_MOD_ID(current->ipc_config.id))? its not checking if the comp belongs to the starting pipeline or not that check is just above.